### PR TITLE
Dev: fix jump and error, support terminal cwd

### PR DESCRIPTION
**Pull request by fengwk**
Description:

This pull request includes the following commits:
1. fix: fix cursor set for jdtls
2. fix: use lsp jump instead of vim edit to open jdt:// protocol buffer
3. feat: supports specifying terminal cwd
4. fix: fix error caused by invalid buf， maybe caused by defer function

Please review the changes. Thank you!

### CODE REVIEW

**Comment by glepnir** [comment]
LGTM buf need stylua format

**Comment by fengwk** [comment]
I am not familiar with the Lua specifications. I made some adjustments according to the lint instructions. I think it should be like this?

File: lua/lspsaga/callhierarchy.lua [hunk 1]
```diff
@@ -245,17 +245,23 @@ function ch:keymap()
         return
       end
       local data = self.method == get_method(2) and curnode.value.from or curnode.value.to
-      local fname = vim.uri_to_fname(data.uri)
       local start = data.selectionRange.start
-      local pos = { start.line + 1 }
       self:clean()
       local restore = win:minimal_restore()
-      vim.cmd[action](fname)
+      vim.cmd(action)
```
Comment (glepnir) [inline lua/lspsaga/callhierarchy.lua:251]: could we do action with file name ? not action then use jump_to_location to open file if that function failed the file not open right ?

Comment (glepnir) [reply lua/lspsaga/callhierarchy.lua:251]: fair enough .

**Pull request merged**